### ISSUE TITLE
`MD051` false positive: Link fragments for headings containing colons (`:`) are flagged as invalid

### ISSUE THREAD

**Issue by zts2005** [DavidAnson/markdownlint#1959]
**What did you do?**
I created a Markdown document with a heading containing an English colon (`:`) and tried to link to it using a fragment generated by removing the punctuation, which follows the GitHub heading algorithm.

Example code:

```markdown
## some:thing

[link](#something)

```

**What did you expect to happen?**
The link `[link](#something)` should be considered valid. According to the GitHub heading algorithm (and how VS Code's built-in preview/autocomplete works), punctuation like colons should be removed when generating the fragment. Therefore, `## some:thing` should result in the ID `something`.

**What actually happened?**
The linter triggers a `MD051/link-fragments` violation: "Link fragments should be valid".
Interestingly, if I change the colon to a Chinese colon (`：`) or remove it entirely, the warning disappears.

**What messages or errors were there?**
`MD051/link-fragments: Link fragments should be valid`

**How can the issue be reproduced?**
Paste the following content into a Markdown file with `MD051` enabled:

```markdown
## some:thing

[link](#something)

```

The linter will flag `#something` as invalid.

**What version were you using?**

* VS Code `markdownlint` extension version: 0.61.1

**What operating system were you using?**
Windows 10 22H2

**Comment by zts2005** [comment]
Oh, I think I should have submitted the issue in the main repository, but it seems I made a mistake.

I'm very sorry.

**Comment by DavidAnson** [comment]
You can see from the rendered output in the demo app that the text after the colon in your scenario is being recognized as a directive which is what causes this problem. You can read more about directives here:

https://github.com/micromark/micromark-extension-directive

This problem can be avoided by escaping the colon in the title with a backlash character as shown here:

https://dlaa.me/markdownlint/#%25m%23%20Issue%201959%0A%0A%23%23%20some%3Athing%0A%0A%5Btest%20link%5D(%23something)%0A%0A%23%23%20other%5C%3Athing%0A%0A%5Btest%20link%5D(%23otherthing)%0A

**Comment by zts2005** [comment]
Okay, I understand. This is the first time I have learned about this feature. Thank you.
